spec/exmples/api_spec.rb

```diff
@@ -0,0 +1,89 @@
+# require File.expand_path(File.dirname(__FILE__) + '../../spec_helper')
+
+# class MockOAuth2Error < StandardError
+# end
+
+describe "Api" do
+    before(:each) do
+      @oauth = double("oauth")
+      allow(@oauth).to receive(:get).and_return("get response")
+      allow(@oauth).to receive(:post).and_return("post response")
+      allow(@oauth).to receive(:put).and_return("put response")
+      @api = GoogleContactsApi::Api.new(@oauth)
+    end
+
+    describe ".get" do
+      it "should perform a get request using oauth returning json with version 3" do
+        # expectation should come before execution
+        expect(@oauth).to receive(:get).with(
+          GoogleContactsApi::Api::BASE_URL + "any_url?alt=json&param=param&v=3", {"header" => "header"})
+        expect(@api.get("any_url",
+          {"param" => "param"},
+          {"header" => "header"})).to eq("get response")
+      end
+
+      it "should perform a get request using oauth with the version specified" do
+        expect(@oauth).to receive(:get).with(
+          GoogleContactsApi::Api::BASE_URL + "any_url?alt=json&param=param&v=2", {"header" => "header"})
+        expect(@api.get("any_url",
+          {"param" => "param", "v" => "2"},
+          {"header" => "header"})).to eq("get response")
+      end
+    end
+
+    describe ".post_v2" do
+      it "should perform a post request using oauth" do
```

Comment:
Yeah you don't want to connect to google in your tests.
You _do_ want to stub out and test scenarios where google sends you an Unauthorized response.